mitm.go

```diff
@@ -187,27 +185,17 @@ func (ic *Interceptor) initCrypto() (err error) {
 	}
 	ic.issuingCertPem = ic.issuingCert.PEMEncoded()
 	if ic.opts.InstallCert {
-		isInstalled, _ := ic.issuingCert.IsInstalled()
-		if !isInstalled {
-			var installErr error
-			if ic.opts.InstallCertResult != nil {
-				defer func() {
-					ic.opts.InstallCertResult(installErr)
-				}()
-			}
-			if runtime.GOOS == "windows" && ic.opts.WindowsPromptTitle != "" && ic.opts.WindowsPromptBody != "" {
-				cmd := exec.Command("mshta", fmt.Sprintf("javascript: var sh=new ActiveXObject('WScript.Shell'); sh.Popup('%v', 0, '%v', 64); close()", ic.opts.WindowsPromptBody, ic.opts.WindowsPromptTitle))
-				promptErr := cmd.Run()
-				if promptErr != nil {
-					installErr = fmt.Errorf("Unable to show windows prompt for installing certificate: %v", promptErr)
-					return installErr
-				}
-			}
-			err = ic.issuingCert.AddAsTrustedRoot(ic.opts.InstallPrompt)
-			if err != nil {
-				installErr = fmt.Errorf("Unable to install issuing cert: %v", err)
-				return installErr
-			}
+		var installErr error
+		if ic.opts.InstallCertResult != nil {
+			defer func() {
+				ic.opts.InstallCertResult(installErr)
```

Comment:
In practice, we use this to log whenever we attempted to install a cert. The way you have it now, this will get called every time Lantern runs, regardless of whether or not we attempted to install a cert.

Reply:
@oxtoacart  hmmm... so would it make sense to add a bool return value to AddAsTrustedRootIfNeeded to indicate if any changes were actually made?

Reply:
I think what we need is a bool to indicate whether we attempted to install a certificate.

Reply:
That's what I meant. :) Good. Will do

Reply:
That would imply we need to add isIntalled checks on Linux too so that we only attempt to install if necessary.

Reply:
@oxtoacart - but then we are back to where we started with Linux.
if IsInstalled will check all the nss locations, it will return 'true' when atleast one has the cert installed and then we will not handle the other browsers. That's why I removed the IsInstalled from Linux

Reply:
What I mean is that we should check each location before installing, only install if it's not installed at that location, and then return true if any of the locations required installation.

```
didInstall = false
for location in locations {
    if !isInstalledAt(location) {
        installAt(location)
        didInstall = true
    }
}
return didInstall
}

Reply:
Or alternately, you could allow the caller to installIfNeeded to supply an `onInstallAttempted` callback function